checkfield/checkfield.go

```diff
@@ -72,13 +101,13 @@ func CheckImmutableFields(msgReq *interface{}, msgUpdate *interface{}, immutable
 
 // CheckResourceID implements follows https://google.aip.dev/122#resource-id-segments
 func CheckResourceID(id string) error {
-	if match, _ := regexp.MatchString("^[a-z]([a-z0-9-]{0,61}[a-z0-9])?$", id); !match {
-		return status.Error(codes.InvalidArgument, "The id of pipeline needs to be within ASCII-only 63 characters following RFC-1034 with a regexp (^[a-z]([a-z0-9-]{0,61}[a-z0-9])?$)")
-	}
 	if _, err := uuid.Parse(id); err == nil {
-		return status.Error(codes.InvalidArgument, "The id is a UUID, which is not allowed")
+		return status.Error(codes.InvalidArgument, "`id` is not allowed to be a UUID")
```

Comment:
Actually, we don't have to check whether the id is a UUID, since a UUID won't match the RFC-1034 Regexp anyway. So do we remove the UUID checking?

Reply:
I think UUID is a subset of RFC-1034. Check [this](https://regex101.com/r/2RRStv/1).

Reply:
Ok, we can't remove this. some UUID matches RFC-1034, and some others don't. Check [this](https://regex101.com/r/QE2p2u/1). So it is necessary to retain this check.